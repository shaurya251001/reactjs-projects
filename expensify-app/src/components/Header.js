import React from "react";
import { BrowserRouter, Route, Switch, Link, NavLink } from "react-router-dom";

const Header = () => {
  return (
    <header>
      <h1>Expensify</h1>
      <NavLink to="/" activeClassName="isactive" exact={true}>
        Dashboard
      </NavLink>
      <NavLink to="/create" activeClassName="isactive">
        Create Expense
      </NavLink>
    </header>
  );
};

export default Header;
