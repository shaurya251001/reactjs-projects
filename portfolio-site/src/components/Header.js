import React from "react";
import { NavLink, Switch } from "react-router-dom";

const Header = () => {
  return (
    <header>
      <h1>Portfolio</h1>

      <NavLink activeClassName="isactive" to="/" exact={true}>
        Home
      </NavLink>
      <NavLink activeClassName="isactive" to="/portfolio">
        Portfolio
      </NavLink>
      <NavLink activeClassName="isactive" to="/contact">
        Contact
      </NavLink>
    </header>
  );
};

export default Header;
