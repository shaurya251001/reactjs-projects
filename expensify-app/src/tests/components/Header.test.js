//react-test-renderer
import ReactShallowRenderer from "react-test-renderer/shallow";
import React from "react";
import Header from "../../components/Header";

test("Should set header correctly", () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Header />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
