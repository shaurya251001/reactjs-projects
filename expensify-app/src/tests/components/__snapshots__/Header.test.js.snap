// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should set header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="isactive"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="isactive"
    ariaCurrent="true"
    to="/create"
  >
    Create Expense
  </NavLink>
</header>
`;
